470 - Flatten Download folder

nozomione opened the pull request:
## Issue Number

Close #470

## Purpose/Implementation Notes

I've flattened the `Download` folder and updated all relevant imports across the codebase to reflect the new structure.

## Types of changes

- Refactor (addresses code organization and design mentioned in corresponding issue)

## Functional tests

N/A

## Checklist

<!-- Put an `x` in the boxes that apply. -->

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

N/A

nozomione commented on src/components/DownloadEmpty.js:
```diff
@@ -3,7 +3,7 @@ import { Box, Heading } from 'grommet'
 import { Button } from 'components/Button'
 
-export const EmptyDataset = () => {
+export const DownloadEmpty = () => {
```
I've renamed and prefixed this component with `Download` as this is used for the empty state of the download page.

davidsmejia commented on src/components/FilesSummary.js:
Suggested name `DatasetFileSummaries` and internally instead of `Card` can we use `DatasetFileSummary`. Broadly speaking no one looks in the middle of the name to look for plurality so I would strongly suggest always making the ends of "listing" components plural despite English grammar.

davidsmejia commented on src/components/DownloadOptionsForm.js:
suggested name `DatasetDownloadOptionsForm`

davidsmejia commented on src/components/DatasetDetails.js:
suggested name `DatasetSamples`

davidsmejia commented on src/components/DatasetDetails.js:
```diff
@@ -15,11 +15,21 @@ export const DatasetDetails = ({ dataset, isImmutable }) => {
       label: 'Species View',
-      component: <SpeciesView dataset={dataset} isImmutable={isImmutable} />
+      component: (
+        <DatasetDetailsSpeciesView
```
```suggestion
        <DatasetSamplesOrganisms
```

davidsmejia commented on src/components/DatasetDetails.js:
```diff
@@ -15,11 +15,21 @@ export const DatasetDetails = ({ dataset, isImmutable }) => {
       label: 'Experiment View',
-      component: <ExperimentView dataset={dataset} isImmutable={isImmutable} />
+      component: (
+        <DatasetDetailsExperimentView
```
```suggestion
        <DatasetSamplesExperiments
```

davidsmejia commented on src/components/DatasetDetailsViewBlocks.js:
Can we remove this at this time and just apply the styles directly to the child components.

davidsmejia commented on src/components/StartProcessing.js:
suggested name: `DatasetStartProcessing` and then internally `DatasetStartProcessingForm`

davidsmejia reviewed (commented):
I think this is a good opportunity to update some names. Structurally this is correct so no new files or deletions but lets remove `Download` prefix unless it is a download button for example.

nozomione commented:
I've updated the components with the suggested names and removed the `DatasetDetailsViewBlocks.js` file. This PR is ready for another look. Thank you, David!

davidsmejia reviewed (approved):
LGTM 🚀
